src/niri.rs

```diff
@@ -647,7 +662,60 @@ impl State {
         // build up (the 1 second frame callback timer will call this line).
         self.niri.advance_animations();
 
-        self.niri.redraw_queued_outputs(&mut self.backend);
+        let mut queued_outputs = self.niri.queued_outputs();
+        queued_outputs.sort_by_key(|(_, target_presentation_time, _)| *target_presentation_time);
+
+        let mut min_next_schedule: Option<Duration> = None;
+
+        for (output, target_presentation_time, refresh_interval) in queued_outputs {
+            let next_deadline = self.signal_commit_timing(&output, target_presentation_time);
```

Comment:
Should this happen at the start of `refresh()` together with the call to `notify_blocker_cleared()`? As I understand, this signalling clears blockers for the surfaces that now became ready, which means it should happen earlier, so that the commits are considered by `layout.refresh()`.